### ISSUE TITLE
Unable to find helm chart with clusterscaler version 1.25

### ISSUE THREAD

**Issue by nvithlani** [kubernetes/autoscaler#6471]
**Which component are you using?**:
Cluster-autoscaler

cluster-autoscaler

**What version of the component are you using?**:
1.24

Component version: cluster autoscaler helm chart v9.27.0 (https://artifacthub.io/packages/helm/cluster-autoscaler/cluster-autoscaler/9.27.0)

**What k8s version are you using (`kubectl version`)?**:

<details><summary><code>kubectl version</code> Output</summary><br><pre>
$ kubectl version

kubectl version
Client Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.4", GitCommit:"e6c093d87ea4cbb530a7b2ae91e54c0842d8308a", GitTreeState:"clean", BuildDate:"2022-02-16T12:38:05Z", GoVersion:"go1.17.7", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"25+", GitVersion:"v1.25.16-eks-77b1e4e", GitCommit:"c39acac7636499dd73442b62e11951592aedd1f8", GitTreeState:"clean", BuildDate:"2024-01-13T01:51:36Z", GoVersion:"go1.20.10", Compiler:"gc", Platform:"linux/amd64"}
WARNING: version difference between client (1.23) and server (1.25) exceeds the supported minor version skew of +/-1
</pre></details>

**What environment is this in?**:
AWS

**What did you expect to happen?**:
Expected to have a helm chart for cluster autoscaler v1.25, but the next version of the helm chart (9.28.0), but v9.28.0 contains autoscaler v1.26 which will be incompatible with our EKS v1.25

**What happened instead?**:
Expected to have a helm chart for cluster autoscaler v1.25, but the next version of the helm chart (9.28.0), but v9.28.0 contains autoscaler v1.26 which will be incompatible with our EKS v1.25

We would like to upgrade our cluster autoscaler version to 1.25 using a helm chart
**How to reproduce it (as minimally and precisely as possible)**:
The helm chart repo shows the app versions
https://artifacthub.io/packages/helm/cluster-autoscaler/cluster-autoscaler/9.27.0
https://artifacthub.io/packages/helm/cluster-autoscaler/cluster-autoscaler/9.28.0

**Anything else we need to know?**:
No

**Comment by Shubham82** [comment]
I checked the helm chart versions, I think we missed it because `cluster-autoscaler-chart-9.25.0` to `cluster-autoscaler-chart-9.27.0`
has app version 1.24.0 (CA version)

**Comment by Shubham82** [comment]
cc @gjtempleton
PTAL

**Comment by Shubham82** [comment]
Hi @gjtempleton could you please take a look

**Comment by fmuyassarov** [comment]
I don't know what was the motivation behind jumping to 1.26  from 1.24, but it was done in https://github.com/kubernetes/autoscaler/pull/5651 (I think). Is it too late to update the `appVersion` to 1.25 in the release branch @Shubham82 ?

**Comment by Shubham82** [comment]
@fmuyassarov Yeah, I agree with you, but it would be good to mention this thing under the [README.md](https://github.com/kubernetes/autoscaler/blob/master/charts/cluster-autoscaler/README.md) as a Note so that users can know about this.
WDYT?

**Comment by Shubham82** [comment]
@fmuyassarov, I mean if can't update the app version to the previous version then at least we have to mention this thing under the README.md as a note.
The helm chart version or patch version is released whenever we update/add anything under the Charts. So suppose we change the app version to v1.25 (current app version v1.29.0) then the Chart release would be 9.36.0 (current chart version 9.35.0) and I don't think that is right.

@gjtempleton could you please take a look?

**Comment by fmuyassarov** [comment]
> @fmuyassarov, I mean if can't update the app version to the previous version then at least we have to mention this thing under the README.md as a note. The helm chart version or patch version is released whenever we update/add anything under the Charts. So suppose we change the app version to v1.25 (current app version v1.29.0) then the Chart release would be 9.36.0 (current chart version 9.35.0) and I don't think that is right.
> 
> @gjtempleton could you please take a look?

Makes sense. But I wonder if release mechanism in the project could be improved to avoid such issues in the future.

**Comment by gjtempleton** [comment]
Hey,

This has historically been a gap in our release process for the CA, particularly when we release the new 1.X.0 version of the CA, there's currently no requirement/instructions to also release a new version of the Helm chart setting the app version & image tag to the new version of the CA.

I'm going to discuss with the rest of the release team and propose that we add this as the final step of cutting a first release for a new minor version of the CA.

**Comment by Shubham82** [comment]
> I'm going to discuss with the rest of the release team and propose that we add this as the final step of cutting a first release for a new minor version of the CA starting with the 1.30.0 release.

Thanks, @gjtempleton, It will be helpful if we release a new minor version of the Helm Chart at the time of releasing a minor version of CA.
